features/step_definitions/flash_steps.rb

```diff
@@ -1,3 +1,11 @@
+# frozen_string_literal: true
+
 Then /^I should see the flash message "([^"]*)"$/ do |flash_text|
-  find(:xpath, "//div[@id='flash-messages'] | //div[@id='flashWrapper']").text.should have_content(flash_text)
+  assert_flash(flash_text)
+end
+
+def assert_flash(flash_text)
+  # Sometimes the flash hides before the a test asserts it and fails. Therefore, we target that the flash element
+  # is rendered with the proper message, regardless of visibility.
```

Comment:
do you know why?

Reply:
I am not sure but it seems to be the test browser is a bit slow and it takes a while to render. It's like the page is rendered after a few seconds and the flash is displayed and hidden at the same time, before this step is used.
I haven't found a reliable way to reproduce this... it seems to happen only in Circleci but not when running with ssh.

Reply:
we use small runners so being slow is expected :/
On the other hand, with oracle the runner is big. But then oracle is heavy.